client/src/components/Verejne/Map/index.js

```diff
@@ -80,11 +85,27 @@ export default compose(
       zoomToLocation,
     }
   ),
+  withRouter,
   withDataProviders(({entitiesUrl}) => [entitiesProvider(entitiesUrl)]),
   withHandlers({
-    onChange: ({entitiesUrl, setMapOptions, refetch}) => (options) => {
-      setMapOptions(options)
-    },
+    onChange: (props) => (options) => {
+      console.log(props)
+      props.setMapOptions(options)
+      props.history.push(
```

Comment:
replace instead of pushing , otherwise the person using the app will have to go back through all of the map changes when navigating via back button